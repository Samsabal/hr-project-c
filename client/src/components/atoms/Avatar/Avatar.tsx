import React from "react";
import { getInitials } from "../../../utils/textManipulation";

export const Avatar = ({ name, color }: { name: string; color: "primary" | "gray" }) => {
    return (
        <>
            {color === "primary" ? (
                <div className='h-12 w-12 bg-primary-100 dark:bg-primary-400 dark:bg-opacity-10 rounded-full flex items-center justify-center text-xl text-primary-600 dark:text-primary-500 font-medium'>
                    {getInitials(name)}
                </div>
            ) : (
                <div className='h-12 w-12 bg-gray-100 dark:bg-dark-600 rounded-full flex items-center justify-center text-xl text-gray-600 dark:text-gray-300 font-medium'>
                    {getInitials(name)}
                </div>
            )}
        </>
    );
};
