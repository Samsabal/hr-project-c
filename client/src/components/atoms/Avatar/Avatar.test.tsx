import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import { Avatar } from "./Avatar";

describe ("Avatar", () => {
    test('renders avatar component correctly and has all props', () => {
        const container1 = render(
            <Avatar
            name="John Doe"
            color="primary"
            />
        );
        expect(container1).toMatchSnapshot();
        
        const container2 = render(
            <Avatar
            name="Jane Doe"
            color="gray"
            />
        );
        expect(container2).toMatchSnapshot();
    });
});