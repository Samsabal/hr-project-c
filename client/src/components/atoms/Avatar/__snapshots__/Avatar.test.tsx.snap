// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders avatar component correctly and has all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-12 w-12 bg-primary-100 dark:bg-primary-400 dark:bg-opacity-10 rounded-full flex items-center justify-center text-xl text-primary-600 dark:text-primary-500 font-medium"
      >
        JD
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-12 w-12 bg-primary-100 dark:bg-primary-400 dark:bg-opacity-10 rounded-full flex items-center justify-center text-xl text-primary-600 dark:text-primary-500 font-medium"
    >
      JD
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Avatar renders avatar component correctly and has all props 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-12 w-12 bg-primary-100 dark:bg-primary-400 dark:bg-opacity-10 rounded-full flex items-center justify-center text-xl text-primary-600 dark:text-primary-500 font-medium"
      >
        JD
      </div>
    </div>
    <div>
      <div
        class="h-12 w-12 bg-gray-100 dark:bg-dark-600 rounded-full flex items-center justify-center text-xl text-gray-600 dark:text-gray-300 font-medium"
      >
        JD
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-12 w-12 bg-gray-100 dark:bg-dark-600 rounded-full flex items-center justify-center text-xl text-gray-600 dark:text-gray-300 font-medium"
    >
      JD
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
